hostsfile/src/lib.rs

```diff
@@ -284,6 +284,44 @@ impl HostsBuilder {
     fn write_and_swap(temp_path: &Path, hosts_path: &Path, contents: &[u8]) -> io::Result<()> {
         // Copy the file we plan on modifying so its permissions and metadata are preserved.
         std::fs::copy(hosts_path, temp_path)?;
+
+        #[cfg(feature = "selinux")]
+        if selinux::current_mode() != selinux::SELinuxMode::NotRunning {
+            log::debug!("SELinux is running; copying context");
+            use selinux::SecurityContext;
+            use std::{fs::File, os::fd::AsRawFd};
+
+            let hosts = File::open(hosts_path)?;
+            let temp = File::options().append(true).open(temp_path)?;
+            match SecurityContext::of_file(&hosts.as_raw_fd(), false) {
+                Ok(Some(context)) => {
+                    log::debug!(
+                        "{} context is {:?}",
+                        hosts_path.display(),
+                        context.to_c_string()
+                    );
+                    if let Err(err) = context.set_for_file(&temp.as_raw_fd()) {
```

Comment:
Can we use [`of_path`](https://docs.rs/selinux/latest/selinux/struct.SecurityContext.html#method.of_path) and [`set_for_path`](https://docs.rs/selinux/latest/selinux/struct.SecurityContext.html#method.set_for_path) here?

```suggestion
            match SecurityContext::of_path(hosts_path, false) {
                Ok(Some(context)) => {
                    log::debug!(
                        "{} context is {:?}",
                        hosts_path.display(),
                        context.to_c_string()
                    );
                    if let Err(err) = context.set_for_path(temp_path) {
```

Reply:
ah didn't notice those functions were available, done!